backend/courses/views.py

```diff
@@ -376,3 +384,122 @@ def get_queryset(self):
             section__course__semester=get_current_semester(),
             in_add_drop_period=True,
         ).order_by("created_at")
+
+
+# private helper function used to get user's accepted friends
+def get_accepted_friends(user):
+    friendships = Friendship.objects.filter(Q(sender=user) | Q(recipient=user))
+    accepted_friendships = friendships.filter(status=Friendship.FriendshipStatus.ACCEPTED)
+    return [model_to_dict(friendship) for friendship in accepted_friendships]
+
+class FriendshipViewSet(viewsets.ViewSet):
+    model = Friendship
+    queryset = Friendship.objects.all()
+    http_method_names = ["get", "post", "delete"]
+    permission_classes = [IsAuthenticated]
+
+    # TODO: generate proper PcxAutoschema for this viewset
+    schema = PcxAutoSchema()
+
+    def get(self, request):
+        # get user's friends and friend requests
+        res = {}
+        user = request.user
+
+        friendships = self.queryset.filter(Q(sender=user) | Q(recipient=user))
+        accepted_friendships = friendships.filter(status=Friendship.FriendshipStatus.ACCEPTED)
+        pending_friendships = friendships.filter(status=Friendship.FriendshipStatus.SENT)
+        rejected_friendships = friendships.filter(status=Friendship.FriendshipStatus.REJECTED)
+        res["accepted"] = [model_to_dict(friendship) for friendship in accepted_friendships]
+        res["pending"] = [model_to_dict(friendship) for friendship in pending_friendships]
+        res["rejected"] = [
+            model_to_dict(friendship) for friendship in rejected_friendships
+        ]  # unsure if we should include this
+        res["message"] = "Friendships retrieved successfully."
+        return JsonResponse(res)
+                
+    def post(self, request):
+        res = {}
+        # send friendship request to passed in recipient
+        sender = request.user
+        recipient = get_object_or_404(User, id=request.recipient_id)
+
+        if not recipient:
+            res["message"] = "Recipient does not exist."
+            return JsonResponse(res, status=400)
+
+        # if friend request already exists but was rejected, we can reinstate it:
+        friend_req = self.queryset.get(sender=sender, recipient=recipient, status=Friendship.FriendshipStatus.REJECTED)
```

Comment:
Might want to just get any existing friendship regardless of status (only making 1 trip to the db), and then case on the status once it's in-memory. We could do:
```python
existing_friendship = self.queryset.filter(
    Q(recipient=recipient) | Q(sender=recipient)
).first()
```

Using `first` is important because `get` will throw an exception if the object does not exist. `first` will just return `None`.